### ISSUE TITLE
support flexible learning rate adjustment strategy

### ISSUE THREAD

**Issue by jacquesqiao** [PaddlePaddle/Paddle#7769]
- [x] learning rate should be a global variable. depends on https://github.com/PaddlePaddle/Paddle/issues/7784
- [x] define the interface to do learning rate decay.  [pd-refs](http://www.paddlepaddle.org/docs/develop/documentation/zh/faq/parameter/index_cn.html#permalink-2-2-learning-rate-annealing-) [tf-refs](https://www.tensorflow.org/versions/r0.12/api_docs/python/train/decaying_the_learning_rate):  
  - [x] framework https://github.com/PaddlePaddle/Paddle/pull/7892
  - [x] exponential_decay  https://github.com/PaddlePaddle/Paddle/pull/7892
  - [x] inverse_time_decay #7892 
  - [x] piecewise_decay #8013 
  - [x] polynomial_decay #8013
- [x] use condition op to adjust the learning rate during training. #8013 #8184 
- [x] add elementwise_pow_op https://github.com/PaddlePaddle/Paddle/pull/7911
- [x] support syntax like `a+b, a*10`. https://github.com/PaddlePaddle/Paddle/pull/7688
- [x] fix floor op https://github.com/PaddlePaddle/Paddle/pull/7926
- [x] what device should learing rate decay happen? should it all happen on CPU? https://github.com/PaddlePaddle/Paddle/pull/8254
- [x] Add switch op #8184

**Comment by reyoung** [comment]
This issue depends on #7784

**Comment by jacquesqiao** [comment]
all task is done now
